### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by alexanderattar**
## Describe Changes

- Add table of contents, architecture, and spec sections

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All Hardhat tests are passing

### CODE REVIEW

**Review by ACC01ADE — APPROVED** [review]
Did not do any spell checks since this is the content from the docs.holograph.xyz site.

**Comment by alexanderattar** [comment]
> any spell checks since this is the content from the docs.holograph.xyz site.

Edited this in a markdown editor with spellcheck enabled so we should be good

File: README.md [hunk 1]
```diff
@@ -6,10 +6,96 @@
...
+
+- [Description](#description)
+- [Specification](#specification)
+- [Architecture](#architecture)
+- [Development](#development)
+- [Directories](#directory-structure)
+- [Branching](#branching-model-and-releases)
+- [Contributing](#contributing)
+- [Links](#official-links)
+- [License](#license)
```
Comment (sogoiii) [inline README.md:19]: I love this.

**Review by sogoiii — APPROVED** [review]
Should the table of networks and address information be in this readme or should we only add it to the documentation site?

**Pull request merged**